Refactor RgbdCamera

kunimatsu-tri opened the pull request:
Refactor `RgbdCamera` by making use of `RgbdRenderer` which is introduced by #7228.

In this PR, there is no newly added functionality.  Just refactoring `RgbdCamera`, getting rid of duplicated unit tests with `RgbdRenderer`'s and cleaning up the code.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7269)
<!-- Reviewable:end -->

kunimatsu-tri commented:
+@thduynguyen feature review please.

cc/ @EricCousineau-TRI

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-KwazjucL3AunRgfInBS:b-ozc6o8)*
<!-- Sent from Reviewable.io -->

thduynguyen commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[drake/systems/sensors/rgbd_camera.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb0jDTIoyNVeTynaBT:-Kwb0jDTIoyNVeTynaBU:bwg5izu) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L21)):*
> ```cpp
> namespace {
> 
> const int kPortStateInput = 0;
> ```

BTW Should it be constexpr?
Also a comment for this constant.

---

*[drake/systems/sensors/rgbd_camera.cc, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb1-Ih09hR1F35Ierd:-Kwb1-Ih09hR1F35Iere:bu9vsrr) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L24)):*
> ```cpp
> 
> // TODO(kunimatsu-tri) Add support for the arbitrary image size
> const int kImageWidth = 640;  // In pixels
> ```

BTW constexpr?

---

*[drake/systems/sensors/rgbd_camera.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb115VlaZosOFmlaXl:-Kwb115VlaZosOFmlaXm:bu9vsrr) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L25)):*
> ```cpp
> // TODO(kunimatsu-tri) Add support for the arbitrary image size
> const int kImageWidth = 640;  // In pixels
> const int kImageHeight = 480;  // In pixels
> ```

BTW constexpr?

---

*[drake/systems/sensors/rgbd_camera.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb15EAG2WAYk8QnGkp:-Kwb15EAG2WAYk8QnGkq:babuavv) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L27)):*
> ```cpp
> const int kImageHeight = 480;  // In pixels
> 
> Eigen::Isometry3d X_BC() {
> ```

BTW Why does it need to be a function? Smells like another constexpr to me.

---

*[drake/systems/sensors/rgbd_camera.cc, line 30 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb1NfQ-ZCPt9-MlaO3:-Kwb1NfQ-ZCPt9-MlaO4:b99fnuh) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L30)):*
> ```cpp
>   // The color sensor's origin (`Co`) is offset by 0.02 m on the Y axis of
>   // the RgbdRenderer's base coordinate system (`B`).
>   // TODO(kunimatsu-tri) Add support for arbitrary relative pose.
> ```

BTW You mean as a member variable in RgbdCamera?

---

*[drake/systems/sensors/rgbd_camera.cc, line 58 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwehKn3swai96hEdv0H:-KwehKn3swai96hEdv0I:b-ca5ox0) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L58)):*
> ```cpp
>       if (z == RgbdRenderer::InvalidDepth::kTooClose ||
>           z == RgbdRenderer::InvalidDepth::kTooFar) {
>         pc(0, v * width + u) = RgbdRenderer::InvalidDepth::kTooFar;
> ```

BTW Should we use another constant for this as InvalidDepth::kTooFar comes with a surprise here? Something like kOutOfRangeOutput?

---

*[drake/systems/sensors/rgbd_camera.cc, line 66 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb2xgElQqPvYpJUy5v:-Kwb2xgElQqPvYpJUy5w:bmobbu1) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L66)):*
> ```cpp
>         pc(2, v * width + u) = z;
>       }
>     }
> ```

BTW, Not 100% sure, but I think it could be bit faster if we do something like `pc = Eigen::Matrix3Xf::Constant(3,size, kTooFar)` and only assign depth for valid pixels... Doesn't matter.

---

*[drake/systems/sensors/rgbd_camera.cc, line 87 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb3yZUdfy9wyxOb_kn:-Kwb3yZUdfy9wyxOb_ko:b-pcwqbd) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L87)):*
> ```cpp
>       // TODO(kunimatsu-tri) Change the X_BD_ to be different from X_BC_ when
>       // it's needed.
>       X_BC_(X_BC()), X_BD_(X_BC()),
> ```

BTW There's no API to let users initialize these values. Ideally, they should be read from a config (urdf?) file?

---

*[drake/systems/sensors/rgbd_camera.cc, line 110 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb4__z2XinlwkhOR9Z:-Kwb4__z2XinlwkhOR9_:bc97arz) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L110)):*
> ```cpp
>       // TODO(kunimatsu-tri) Change the X_BD_ to be different from X_BC_ when
>       // it's needed.
>       X_BC_(X_BC()), X_BD_(X_BC()),
> ```

BTW As above

---

*[drake/systems/sensors/rgbd_camera.cc, line 199 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb5iTiF7AAbr-KoafJ:-Kwb5iTiF7AAbr-KoafK:b-2llgcp) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L199)):*
> ```cpp
> 
>   Eigen::Translation<double, 3> trans =
>       Eigen::Translation<double, 3>(X_WB.translation());
> ```

BTW I think using {} makes it easier to read.

---

*[drake/systems/sensors/rgbd_camera.cc, line 201 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweYi_FG9V5BVPWVdgV:-KweYi_FG9V5BVPWVdgW:bm9vttx) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L201)):*
> ```cpp
>       Eigen::Translation<double, 3>(X_WB.translation());
>   pose_vector->set_translation(trans);
>   Eigen::Quaterniond quat = Eigen::Quaterniond(X_WB.linear());
> ```

BTW Same as above.

---

*[drake/systems/sensors/rgbd_camera.h, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwe_hAuHJY4xCu6DBq-:-Kwe_hAuHJY4xCu6DBq0:b-kylsat) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L26)):*
> ```objc
> /// backend.
> /// Its image resolution is fixed at VGA (640 x 480 pixels) for all three
> /// images. The default depth sensing range is 0.5 m to 5.0 m.
> ```

BTW "is from .. to .."

---

*[drake/systems/sensors/rgbd_camera.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweZ_l6qYE5ee1N9CD5:-KweZ_l6qYE5ee1N9CD6:b-on41ix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L64)):*
> ```objc
> ///
> /// @ingroup sensor_systems
> class RgbdCamera final : public LeafSystem<double> {
> ```

BTW Wonder if/should the whole class be templated on the scalar type. Maybe it's not important.

---

*[drake/systems/sensors/rgbd_camera.h, line 70 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweaxBT_cid4ibgYh5-:-KweaxBT_cid4ibgYh50:br8wlk2) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L70)):*
> ```objc
>   /// Converts a depth image obtained from RgbdCamera to a point cloud.  If a
>   /// pixel in the depth image has NaN depth value, all the `(x, y, z)` values
>   /// in the converted point will be NaN.
> ```

BTW I don't see this NaN in the implementation. Is it true that an NaN involved in an expression results in an NaN? Maybe it's worth a comment in the cpp file.

---

*[drake/systems/sensors/rgbd_camera.h, line 97 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwebkUgirLRA0ZaYxrL:-KwebkUgirLRA0ZaYxrM:brq7f7h) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L97)):*
> ```objc
>   /// world. The life span of this parameter must exceed that of this class's
>   /// instance. The maximum number of bodies in the `tree` must be less than
>   /// 1536 based on the number of the colors used for the label image.
> ```

BTW What happen if this condition fails? Do we have an assertion/exception check?

---

*[drake/systems/sensors/rgbd_camera.h, line 222 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwenSGezLJf7XA4UAej:-Kwem2-J3xW-NhjPG-X3:bhhp5rt) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L222)):*
> ```objc
>   void UpdateModelPoses(const BasicVector<double>& input_vector) const;
> 
>   const OutputPort<double>* color_image_port_{};
> ```

BTW Due to the spacing, I mistakenly thought of this as a const pointer at first instead of a pointer to const. Would prefer "OutputPort<double> const *" or "const OutputPort<double> *". Not important. Perhaps it's just my problem and I need to get used to it. :)

---

*[drake/systems/sensors/rgbd_camera.h, line 288 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwep5D0DZC_9wji-gRy:-Kwep5D0DZC_9wji-gRz:bub77et) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L288)):*
> ```objc
> 
>  private:
>   RgbdCamera* const camera_{};
> ```

BTW I don't think this in-class initialization is needed. I'm surprised that this *const* pointer can be initialized again in the constructor. In any case it's a confusion.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwedKgLRHMqB2A4KeuE:-KwedKgLRHMqB2A4KeuF:be1crks) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L38)):*
> ```cpp
> // The following tolerance is used due to a precision difference between Ubuntu
> // Linux and Mac OSX.
> const double kTolerance = 1e-12;
> ```

BTW I prefer using constexpr for these constants if possible.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 233 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kweh2AuS7_3XXZLzJTN:-Kweh2AuS7_3XXZLzJTO:b-ofseld) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L233)):*
> ```cpp
> TEST_F(RgbdCameraDiagramTest, FixedCameraOutputTest) {
>   // RgbdCamera is looking straight down 1m above the ground.
>   const Eigen::Vector3d position(0, 0, 1);
> ```

BTW (0., 0., 1.)?

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 234 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kweh5LUh_WhXK1I8tir:-Kweh5LUh_WhXK1I8tis:blc966h) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L234)):*
> ```cpp
>   // RgbdCamera is looking straight down 1m above the ground.
>   const Eigen::Vector3d position(0, 0, 1);
>   const Eigen::Vector3d orientation(0, M_PI_2, 0);
> ```

BTW Same as above

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 252 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwegokzBj0GSWx3tOAP:-KwegokzBj0GSWx3tOAQ:b-leyqev) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L252)):*
> ```cpp
> TEST_F(RgbdCameraDiagramTest, MovableCameraOutputTest) {
>   // RgbdCamera is looking straight down 1m above the ground.
>   const Eigen::Isometry3d X_WB = Eigen::Translation3d(0, 0, 1) *
> ```

BTW Eigen::Translation3d(0., 0., 1.)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
Review status: 5 of 10 files reviewed at latest revision, 21 unresolved discussions.

---

*[drake/systems/sensors/rgbd_camera.cc, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb0jDTIoyNVeTynaBT:-KwfKZlWfnE7hW2uNF9d:b-wnfmk9) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L21)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Should it be constexpr?
Also a comment for this constant.
</blockquote></details>

Done. Removed.

---

*[drake/systems/sensors/rgbd_camera.cc, line 24 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb1-Ih09hR1F35Ierd:-KwfK_onh6ImPNXt5pkj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L24)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW constexpr?
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.cc, line 25 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb115VlaZosOFmlaXl:-KwfKa70dx0y5mlfqwnn:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L25)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW constexpr?
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.cc, line 30 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb1NfQ-ZCPt9-MlaO3:-KwftYEmO7tiCXtWptAD:b32jia5) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L30)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW You mean as a member variable in RgbdCamera?
</blockquote></details>

Done.

Moved from here to where this is called.

---

*[drake/systems/sensors/rgbd_camera.cc, line 87 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb3yZUdfy9wyxOb_kn:-KwevxLtwSPLoLYWbCDE:b-jnzuik) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L87)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW There's no API to let users initialize these values. Ideally, they should be read from a config (urdf?) file?
</blockquote></details>

I had left a TODO comment to do so at `X_BC()`, but I think here is better than there.
Changed the location of the TODO comment.

Done. 

---

*[drake/systems/sensors/rgbd_camera.cc, line 199 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb5iTiF7AAbr-KoafJ:-KwfsUY-Brh4nsXcKdhl:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L199)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW I think using {} makes it easier to read.
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.cc, line 201 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweYi_FG9V5BVPWVdgV:-Kwft5es4NIqa4FBd6AV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L201)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Same as above.
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.h, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwe_hAuHJY4xCu6DBq-:-KwfP2LAukl1wqPYP6p3:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L26)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW "is from .. to .."
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweZ_l6qYE5ee1N9CD5:-KwfP9Owd295l5C4v-fi:bky6k6l) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L64)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Wonder if/should the whole class be templated on the scalar type. Maybe it's not important.
</blockquote></details>

`RgbdCamera` doesn't support `AutoDiff` nor `symbolic::Expression`, so it's not templated.

---

*[drake/systems/sensors/rgbd_camera.h, line 70 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweaxBT_cid4ibgYh5-:-KwfSDcB_QDvR8Rttmy6:bdcqpgc) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L70)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW I don't see this NaN in the implementation. Is it true that an NaN involved in an expression results in an NaN? Maybe it's worth a comment in the cpp file.
</blockquote></details>

Done.

Yes, it's true that any results of calculation with NaN become NaN.  Added a comment in `.cc` file.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 38 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwedKgLRHMqB2A4KeuE:-KwfrMEOWl8xeSzgNcvP:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L38)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW I prefer using constexpr for these constants if possible.
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 233 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kweh2AuS7_3XXZLzJTN:-KwfrT9NU50Sm15C0OEk:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L233)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW (0., 0., 1.)?
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 234 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kweh5LUh_WhXK1I8tir:-KwfrTgLegHyV9DK70Nj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L234)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Same as above
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 252 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwegokzBj0GSWx3tOAP:-KwfrXOXOA3XxaQOdQVP:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/test/rgbd_camera_test.cc#L252)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Eigen::Translation3d(0., 0., 1.)?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

thduynguyen commented:
Review status: 5 of 10 files reviewed at latest revision, 11 unresolved discussions.

---

*[drake/systems/sensors/rgbd_camera.cc, line 201 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwgE1AwvwO0an5mtHb3:-KwgE1AwvwO0an5mtHb4:brrhffr) ([raw file](https://github.com/robotlocomotion/drake/blob/f11cc058e9cb8ffb8b484d39ced7f371862425b4/drake/systems/sensors/rgbd_camera.cc#L201)):*
> ```cpp
> 
>   Eigen::Translation<double, 3> trans{
>     Eigen::Translation<double, 3>(X_WB.translation())};
> ```

Can we do `Eigen::Translation<double, 3> trans{X_WB.translation()};`?

---

*[drake/systems/sensors/rgbd_camera.cc, line 204 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwgE-fBE4NdnfUz-RME:-KwgE-fBE4NdnfUz-RMF:be8jc95) ([raw file](https://github.com/robotlocomotion/drake/blob/f11cc058e9cb8ffb8b484d39ced7f371862425b4/drake/systems/sensors/rgbd_camera.cc#L204)):*
> ```cpp
>   pose_vector->set_translation(trans);
> 
>   Eigen::Quaterniond quat{Eigen::Quaterniond(X_WB.linear())};
> ```

How about `Eigen::Quaterniond quat{X_WB.linear()};`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
Review status: 5 of 10 files reviewed at latest revision, 11 unresolved discussions.

---

*[drake/systems/sensors/rgbd_camera.cc, line 201 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwgE1AwvwO0an5mtHb3:-KwlVKCoBchblYL7hde0:b-hjmf4c) ([raw file](https://github.com/robotlocomotion/drake/blob/f11cc058e9cb8ffb8b484d39ced7f371862425b4/drake/systems/sensors/rgbd_camera.cc#L201)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

Can we do `Eigen::Translation<double, 3> trans{X_WB.translation()};`?
</blockquote></details>

Oops, now I got it what you meant.

Done.

---

*[drake/systems/sensors/rgbd_camera.cc, line 204 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwgE-fBE4NdnfUz-RME:-KwlVPY1nAVxNtu1YiEf:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/f11cc058e9cb8ffb8b484d39ced7f371862425b4/drake/systems/sensors/rgbd_camera.cc#L204)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

How about `Eigen::Quaterniond quat{X_WB.linear()};`?
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.h, line 97 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwebkUgirLRA0ZaYxrL:-KwlWnHdjUeY2YdvWR_r:b-6x6439) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L97)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW What happen if this condition fails? Do we have an assertion/exception check?
</blockquote></details>

Replaced `DRAKE_DEMAND` with `std::logic_error` exception in `ColorPalette` and documented.

Done.

---

*[drake/systems/sensors/rgbd_camera.h, line 222 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwenSGezLJf7XA4UAej:-KwlbNPU9_aSh-rde5R3:b-ip12mq) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L222)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Due to the spacing, I mistakenly thought of this as a const pointer at first instead of a pointer to const. Would prefer "OutputPort<double> const *" or "const OutputPort<double> *". Not important. Perhaps it's just my problem and I need to get used to it. :)
</blockquote></details>

For "OutputPort const *", I prefer "const OutputPort*" so that we can align all the other `const` variables, e.g. "`const` something&",  in a class definition, which I think is neat.

For "const OutputPort *", based-on `grep` search for `<T>*` v.s. `<T> *`, the former is popular.

So, for the consistency, I'll leave it as-is.

---

*[drake/systems/sensors/rgbd_camera.h, line 288 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwep5D0DZC_9wji-gRy:-KwlZ17Z4uBGIGVfcWXG:b-buzu6k) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L288)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW I don't think this in-class initialization is needed. I'm surprised that this *const* pointer can be initialized again in the constructor. In any case it's a confusion.
</blockquote></details>

Based-on `grep` search for `const` POD member variables in Drake, I see many cases use both in-class initialization and member initializer list together.

So, for the consistency, I'll leave it as-is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

thduynguyen commented:
Reviewed 2 of 5 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/rgbd_camera.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweZ_l6qYE5ee1N9CD5:-Kwmnmp0MuTHDyTjvmYx:b-yjbzaz) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L64)):*
<details><summary><i>Previously, kunimatsu-tri (Kunimatsu Hashimoto) wrote…</i></summary><blockquote>

`RgbdCamera` doesn't support `AutoDiff` nor `symbolic::Expression`, so it's not templated.
</blockquote></details>

BTW Would it support them in the future? I think a generative model is needed for all sensors if we want to use them for estimation. I bet it should be implemented somewhere here, although I'm not 100% sure as I'm not familiar with Drake yet. Anyway, we can revisit this in the future. I just saw a lot of `<double>` and thought it might just support `<float>`.

---

*[drake/systems/sensors/rgbd_camera.h, line 288 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwep5D0DZC_9wji-gRy:-Kwmp-b0DWm9f9VnxCFP:bc2foq4) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L288)):*
<details><summary><i>Previously, kunimatsu-tri (Kunimatsu Hashimoto) wrote…</i></summary><blockquote>

Based-on `grep` search for `const` POD member variables in Drake, I see many cases use both in-class initialization and member initializer list together.

So, for the consistency, I'll leave it as-is.
</blockquote></details>

BTW Hmm... It should cause a compile error: https://stackoverflow.com/questions/14495536/how-to-initialize-const-member-variable-in-a-class 
Just for my curiosity, does it work if we put `RgbdCamera* const camera_{nullptr};`, which I think is effectively the same as `{}`? Maybe they are not the same and the empty `{}` does nothing?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
Review status: 6 of 15 files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/rgbd_camera.cc, line 27 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb15EAG2WAYk8QnGkp:-Kwq2VMHy8JnKEklxd9E:belz0nz) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L27)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Why does it need to be a function? Smells like another constexpr to me.
</blockquote></details>

Done.  Moved to `.h` file, although I couldn't use `constexpr` here.

---

*[drake/systems/sensors/rgbd_camera.cc, line 58 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwehKn3swai96hEdv0H:-Kwq7d_X_mseWMyVZWz4:b2xlo0k) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L58)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Should we use another constant for this as InvalidDepth::kTooFar comes with a surprise here? Something like kOutOfRangeOutput?
</blockquote></details>

Good point.  This made me think about the right place for `RgbdRenderer::InvalidDepth` and `RgbdRenderer::Label`, and moved them to `image.{h, cc}.

Also, `ConvertDepthImageToPointCloud()` should probably go where `PointCloud` class is located and define new constants there as you point out.  I put TODO comments for now.

---

*[drake/systems/sensors/rgbd_camera.cc, line 66 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb2xgElQqPvYpJUy5v:-KwqGWeioIph1JN0ei7C:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L66)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW, Not 100% sure, but I think it could be bit faster if we do something like `pc = Eigen::Matrix3Xf::Constant(3,size, kTooFar)` and only assign depth for valid pixels... Doesn't matter.
</blockquote></details>

Done.

---

*[drake/systems/sensors/rgbd_camera.cc, line 110 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwb4__z2XinlwkhOR9Z:-Kwq3N_gm1jpq8bZ3D7h:b-degg6s) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.cc#L110)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW As above
</blockquote></details>

Done.  Removed this line and initializing with `{}` in `.h` instead.

---

*[drake/systems/sensors/rgbd_camera.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KweZ_l6qYE5ee1N9CD5:-Kwp08PH8ekEXmNUkyKf:bfcfuiv) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L64)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Would it support them in the future? I think a generative model is needed for all sensors if we want to use them for estimation. I bet it should be implemented somewhere here, although I'm not 100% sure as I'm not familiar with Drake yet. Anyway, we can revisit this in the future. I just saw a lot of `<double>` and thought it might just support `<float>`.
</blockquote></details>

There is no concrete plan for supporting scalar type in `RgbdCamera` so far, but yest I agree that we can revisit in the future.

---

*[drake/systems/sensors/rgbd_camera.h, line 288 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-Kwep5D0DZC_9wji-gRy:-KwpxmpmX4u21AiA37aM:bw74nds) ([raw file](https://github.com/robotlocomotion/drake/blob/4d2dc02e8cb1b72cdba49d8874d6e8a7837fde33/drake/systems/sensors/rgbd_camera.h#L288)):*
<details><summary><i>Previously, thduynguyen (Duy-Nguyen Ta) wrote…</i></summary><blockquote>

BTW Hmm... It should cause a compile error: https://stackoverflow.com/questions/14495536/how-to-initialize-const-member-variable-in-a-class 
Just for my curiosity, does it work if we put `RgbdCamera* const camera_{nullptr};`, which I think is effectively the same as `{}`? Maybe they are not the same and the empty `{}` does nothing?
</blockquote></details>

I think that the compile error happens with C++03, but doesn't with C++11.

And `RgbdCamera* const camera_{nullptr};` worked.

I confirmed that empty `{}` and `{nullptr}` are doing the same thing as they are basically initializing by zero.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
Addressed all the comments. PTAL.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-Kwqc6aBVbkux93fsxCd:b-y0j4qv)*
<!-- Sent from Reviewable.io -->

thduynguyen commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 9 files at r4.
Review status: 10 of 15 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-Kwv2emrrjuu5kn1WUsM:bnfp4nl)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
+@sherm1 platform review please. Thanks!

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-Kwv7HC1VaqNEM5JDKv7:bqs0osh)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Platform <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, after handling a few minor BTWs, ptal.

---

Reviewed 5 of 9 files at r1, 1 of 3 files at r3, 9 of 9 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/image.h, line 151 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvbgSAuoSY-m77rG3Y:-KwvbgSAuoSY-m77rG3Z:b-3jsx2k) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/image.h#L151)):*
BTW took me a while to understand how this sentence might relate to InvalidDepth. Can you clarify? I think you mean something like
> Note that in the case that a depth is not measurable, the constants defined
here are not used. Instead we set the depth to NaN.

?

---

*[drake/systems/sensors/image.h, line 164 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvcN_eWy4VO-Tuj7EV:-KwvcN_eWy4VO-Tuj7EW:b-ofkjzf) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/image.h#L164)):*
> ```objc
>   /// it results in less devastating bugs when users fail to check for the
>   /// lower limit being hit and using negative infinity does not prevent users
>   /// from writing bad code.
> ```

BTW great comment!

---

*[drake/systems/sensors/rgbd_camera.h, line 189 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvZmVnHUoNbUI7rI-_:-KwvZmVnHUoNbUI7rI-a:b-x7g1z5) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/rgbd_camera.h#L189)):*
BTW this error message seems obscure. How about something like:
> "RgbdCamera::frame(): A fixed camera does not have an associated frame."

(Probably should be `std::logic_error` though we haven't been enforcing that in Drake.)

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 53 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvcwSci3NhjnTxf-8u:-KwvcwSci3NhjnTxf-8v:bqfbshn) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L53)):*
> ```cpp
> 
>   // The expected focal value is calculated by the equation here:
>   // https://github.com/RobotLocomotion/drake/blob/master/drake/systems/sensors/camera_info.h#L87
> ```

BTW this comment is too brittle -- it will quickly become invalid. Better to repeat the equation here or just cite to the file or something in the file that is much more stable than a line number.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 292 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwveAwgVenObCTLSWF1:-KwveAwgVenObCTLSWF2:b-snoa0z) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L292)):*
> ```cpp
> 
>   // This must be called by all the test cases first.
>   void SetUp(float depth_value) {
> ```

BTW please consider using a name other than `SetUp()` for this method, since that's the Google Test-defined name. It took me a while to realize that you weren't re-invoking the Google Test fixture.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 325 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwveXvqn4srTmNOx38v:-KwveXvqn4srTmNOx38w:b-t6bqh5) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L325)):*
> ```cpp
>           kDepthValue * (v - kDepthHeight * 0.5) / kFocal;
>       ASSERT_NEAR(actual(1), expected_y, kDistanceTolerance);
>       ASSERT_NEAR(actual(2), kDepthValue, kDistanceTolerance);
> ```

BTW why ASSERTs here rather than EXPECTs? Is that because there are so many pixels that you want to stop as soon as you see a bad one? (Reasonable, I'm just checking.) Would be worth a comment here explaining in any case.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-KwvfIyoc3Qc1C-G1LtB:bneggak)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
Thanks @sharm1 for the platform review. Addressed all the comments, PTAL.

---

Review status: 12 of 15 files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/systems/sensors/image.h, line 151 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvbgSAuoSY-m77rG3Y:-KwvqweXweU5MGt3dulv:b-6x6jk7) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/image.h#L151)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW took me a while to understand how this sentence might relate to InvalidDepth. Can you clarify? I think you mean something like
> Note that in the case that a depth is not measurable, the constants defined
here are not used. Instead we set the depth to NaN.

?
</blockquote></details>

Thanks!

Done.

---

*[drake/systems/sensors/image.h, line 164 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvcN_eWy4VO-Tuj7EV:-Kwvpnor5sJOcvdlE5bP:b-hzpioy) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/image.h#L164)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW great comment!
</blockquote></details>

Thanks!  Although I just copied from here: https://github.com/RobotLocomotion/drake/blob/master/drake/systems/sensors/depth_sensor_output.h#L40

So, this comment should go to Liang :)

---

*[drake/systems/sensors/rgbd_camera.h, line 189 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvZmVnHUoNbUI7rI-_:-Kwvpi_MJfn6-WRLdnGR:b-6x6jk7) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/rgbd_camera.h#L189)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW this error message seems obscure. How about something like:
> "RgbdCamera::frame(): A fixed camera does not have an associated frame."

(Probably should be `std::logic_error` though we haven't been enforcing that in Drake.)
</blockquote></details>

Thanks!

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 53 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwvcwSci3NhjnTxf-8u:-KwvlyW2H0U9MPynZOxH:b-qbunxy) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L53)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW this comment is too brittle -- it will quickly become invalid. Better to repeat the equation here or just cite to the file or something in the file that is much more stable than a line number.
</blockquote></details>

That's a good point.  Removed the line number.

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 292 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwveAwgVenObCTLSWF1:-KwvriMwhfM4Mfp20J66:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L292)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW please consider using a name other than `SetUp()` for this method, since that's the Google Test-defined name. It took me a while to realize that you weren't re-invoking the Google Test fixture.
</blockquote></details>

Done.

---

*[drake/systems/sensors/test/rgbd_camera_test.cc, line 325 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-KwveXvqn4srTmNOx38v:-KwvgNPp4dM1M5zqPtQ9:b-1lqz9o) ([raw file](https://github.com/robotlocomotion/drake/blob/e190016a4f732e1c0b266581063c76795c6c549a/drake/systems/sensors/test/rgbd_camera_test.cc#L325)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

BTW why ASSERTs here rather than EXPECTs? Is that because there are so many pixels that you want to stop as soon as you see a bad one? (Reasonable, I'm just checking.) Would be worth a comment here explaining in any case.
</blockquote></details>

Yes, that's correct.

I put the comment at the top of the file since this is true for all the cases when I use `ASSERT`.

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-KwvshIAVQTyzEJc0Dbf:b-zdcist)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Reviewed 3 of 3 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
@kunimatsu-tri, this is ready to merge but needs a squash-merge with a nice comment for the squashed commit so I'll leave that to you.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-KwwWnkD71PALwYpeh0P:b-sc2qgp)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
@drake-jenkins-bot mac-clang-experimental please

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7269#-:-KwzSFEIgxb1pWx7Pmgm:b-ixvtlv)*
<!-- Sent from Reviewable.io -->
